packages/frontend/components/scaffold-eth/Contract/utils.tsx

```diff
@@ -0,0 +1,86 @@
+import { FunctionFragment } from "ethers/lib/utils";
+import ContractData from "~~/contracts/hardhat_contracts.json";
+import { Contract, utils } from "ethers";
+import DisplayVariable from "~~/components/scaffold-eth/Contract/DisplayVariables";
+import { ReadOnlyFunctionForm } from "./ReadOnlyFunctionForm";
+
+type GeneratedContractType = {
+  address: string;
+  abi: [];
+};
+
+// TODO Add checks if the contract is not present at `contracts` directory
+/**
+ * @param chainId - deployed contract chainId
+ * @param contractName - name of deployed contract
+ * @returns object containing contract address and abi
+ */
+const getGeneratedContract = (
+  chainId: string | undefined,
+  contractName: string | undefined | null,
+): GeneratedContractType => {
+  if (!chainId || !contractName) {
+    return { address: "", abi: [] };
+  }
+
+  const contractsAtChain = ContractData[chainId as keyof typeof ContractData];
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  // ! remove hardcoding of index 0
+  const contractData: GeneratedContractType = contractsAtChain[0].contracts[contractName];
+
+  return contractData;
+};
+
+const getAllContractFunctions = (contract: Contract) => {
+  return contract ? Object.values(contract.interface.functions).filter(fn => fn.type === "function") : [];
+};
+
+const getContractVariablesAndNoParamsReadMethods = (
+  contract: Contract,
+  contractMethodsAndVariables: FunctionFragment[],
+) => {
+  return contractMethodsAndVariables.map((fn, index) => {
+    const isQueryableWithNoParams =
+      (fn.stateMutability === "view" || fn.stateMutability === "pure") && fn.inputs.length === 0;
+    if (isQueryableWithNoParams) {
+      return (
+        <DisplayVariable key={`DV_${fn.name}_${index}`} functionFragment={fn} contractAddress={contract.address} />
+      );
+    }
+    return null;
+  });
+};
+
+const getContractReadOnlyMethodsWithParams = (contract: Contract, contractMethodsAndVariables: FunctionFragment[]) => {
+  return contractMethodsAndVariables.map((fn, index) => {
+    const isQueryableWithParams =
+      (fn.stateMutability === "view" || fn.stateMutability === "pure") && fn.inputs.length > 0;
```

Comment:
I hope this is the right way to get all the functions that are read-only and have params more than one :)

Reply:
Looks good to me!

And for the (ToDo) `getContractWriteMethods` it should be:

`(fn.stateMutability !== "view" && fn.stateMutability !== "pure")`, right?

I think we don't need to differentiate between write methods with params AND write methods without params.